### DISCUSSION TITLE
ActiveRecord::RecordNotFound

### DISCUSSION THREAD

**Discussion by trimngo** [excid3/noticed#159] [Q&A] [answered]
Hi
I have a question regarding getting an ActiveRecord::RecordNotFound when trying to render notifications in json form.

My controller function looks like this

```
  def list
    activities = Notification.where(recipient_id: @current_user.id)

    render json: {
        status: :found,
        activities: activities,
    }
  end
```

I get the following error in the logs (TeamUser is a model in my application)

```
ActiveRecord::RecordNotFound (Couldn't find TeamUser with 'id'=265):
app/controllers/activity_controller.rb:5:in `list'
```

I think when it tries to render activities which are notifications, there are associated records that have been previously deleted that it can't find. I look in the documentation and tried adding the following which unfortunately had no effect.

```
class ApplicationJob < ActiveJob::Base
  discard_on ActiveJob::DeserializationError
end
```

I'm trying also to catch the exception using "rescue_from" but then I don't see any notifications. Do you have any suggestions on how to handle this better?
I want to detect that I couldn't find Teamuser id=265 but continue to return as many notification as possible.

Any advice appreciated

Tri

**Comment by trimngo** [comment, marked as answer]
Of course when I ask the question I figure out the answer
What I do is I run .to_json on each notification
For notifications that trigger an exception I don't add that to a new list I'm creating
I return that new filtered list

There is probably a more efficient way to do this and I still need to clean out the database of stale entries which I could use help with

```
  def list
    activities = Notification.where(recipient_id: @current_user.id)

    filt_activities =[]
    for a in activities
      begin
        temp=a.to_json
        filt_activities.append(a)
      rescue ActiveRecord::RecordNotFound
        nil
      end
    end

    render json: {
        status: :found,
        activities: filt_activities,
    }
  end
```

**Comment by excid3** [comment]
I'd also recommend using the association helper so you notifications are destroyed when an associated object is destroyed.

https://github.com/excid3/noticed#associating-notifications
